Use eslint-plugin-jest inside of Jest.

cpojer opened the pull request:
**Summary**
Updates eslint-config-fb-strict to use the new jest plugin. Removed an outdated unnecessary test file. Added globals from Jest to the eslint plugin.

**Test plan**

Using `it.only` should warn.

SimenB commented on examples/jquery/__tests__/displayUser-test.js:
```diff
@@ -1,5 +1,7 @@
 // Copyright 2004-present Facebook. All Rights Reserved.
 
+/* global document */
```
why not env browser?

cpojer replied:
don't really need it if it is only used in two tests in the entirety of Jest. Jest's other tests don't use jsdom, so document isn't available.

SimenB commented on packages/eslint-config-fb-strict/package.json:
```diff
@@ -8,8 +8,9 @@
     "eslint-plugin-babel": "^4.0.0",
-    "eslint-config-fbjs": "^1.0.0"
+    "eslint-plugin-jest": "^18.1.0",
+    "eslint": "^3.3.0"
```
why not sorted?

cpojer replied:
Nuclide sorts things like this now when pressing fn+f5. cc @vjeux.

SimenB replied:
doing `npm install -D something` will sort it alphabetically again, that diff might get annoying

SimenB commented on packages/eslint-plugin-jest/src/index.js:
```diff
@@ -8,11 +8,33 @@
-  'no-exclusive-tests': require('./rules/no-exclusive-tests'),
-  'no-identical-title': require('./rules/no-identical-title'),
+module.exports = {
+  environments: {
```
why not use `env: jest`?

cpojer replied:
This makes it so when using the jest eslint plugin you can get the globals from there. It allows us to update the global variables in Jest and in this plugin without having to send a PR to the "globals" project. Arguably, Jest shouldn't be a global env available through eslint anyway.

SimenB replied:
Are there any plans to drop the global API, and just to `import { test, expect } from 'jest'` or something in the future? Maybe even just `import test from jest` and having `t` in callbacks, like tap or ava?

cpojer replied:
I'll eventually post an issue about this. @vjeux had some ideas as well, we should do it :)

SimenB commented on packages/jest-jasmine2/src/index.js:
```diff
@@ -49,9 +49,7 @@ function jasmine2(
 
   environment.global.test = environment.global.it;
   environment.global.it.only = environment.global.fit;
-  environment.global.test.only = environment.global.fit;
```
why are these removed?

cpojer replied:
look at line #50.

SimenB replied:
Ah, makes sense!

cpojer commented:
ok this is working now. I renamed `no-exclusive-tests` to `no-focused-tests`. I'll add one more lint rule to this.

cc @hramos can you add the lint rule to the docs? It would be great to recommend it to people. I think the README and docs pages in packages/eslint-plugin-jest should be removed from inside the code and just moved into a docs page on the website :)
